codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/RpcV2.kt

```diff
@@ -65,10 +65,14 @@ class RpcV2HttpBindingResolver(
     /**
      * > Responses for operations with no defined output type MUST NOT contain bodies in their HTTP responses.
      * > The `Content-Type` for the serialization format MUST NOT be set.
-     *
-     * TODO But the problem is we _always_ add input/output types in OperationNormalizer! This makes the `no_output_response` test fail.
      */
-    override fun responseContentType(operationShape: OperationShape): String = requestContentType(operationShape)
+    override fun responseContentType(operationShape: OperationShape): String? {
+        val syntheticOutputTrait = operationShape.outputShape(model).expectTrait<SyntheticOutputTrait>()
+        if (syntheticOutputTrait.originalId == null) {
+            return null
+        }
+        return requestContentType(operationShape)
+    }
```

Comment:
Nice. Let's add a comment explaining that when `syntheticOutputTrait.originalId == null`, it implies that the original operation had no output; this is subtle.

Hot take: a breaking change I've always wanted to make is to *not* apply `OperationNormalizer` in the server, we derive no real benefit from it and it just makes our code more complex.